### PR TITLE
Add more rdocs for RedisMemo::Batch and RedisMemo::MemoizeMethod

### PR DESCRIPTION
**Pull request by katyho**
### Summary
Added a few more rdocs for batch and memoize_method. Feel free to edit or suggest edits as needed!

### CODE REVIEW

File: lib/redis_memo/batch.rb [hunk 1]
```diff
@@ -3,9 +3,28 @@
 require_relative 'cache'
 require_relative 'tracer'
 
+  ##
+  # This class facilitates the batching of Redis calls triggered by +memoize_method+
+  # to minimize the number of round trips to Redis.
+  #
+  # - Batches cannot be nested
```
Comment (katyho) [inline lib/redis_memo/batch.rb]: Mostly copy and pasted this from the redis_memo.rb class

File: lib/redis_memo/memoize_method.rb [hunk 2]
```diff
@@ -8,6 +8,28 @@
 require_relative 'util'
 
 module RedisMemo::MemoizeMethod
+
+  # Core entry method for using RedisMemo to cache a method's results. When a method is memoized, all
+  # calls to the method  will first check if the results exist in the RedisMemo cache before calling
+  # the original method.
+  #
+  # @example
```
Comment (katyho) [inline lib/redis_memo/memoize_method.rb:15]: Copy and pasted from the readme as a good example of how to use memoize_method with dependencies

File: lib/redis_memo/batch.rb [hunk 3]
```diff
@@ -3,9 +3,28 @@
...
+#   the actual method value
+# - The actual method values are returned as a list, in the same order as
+#   invoking, after exiting the block
+#
+# @example
+#   results = RedisMemo.batch do
+#     5.times { |i| memoized_calculation(i) }
+#     nil # Not the return value of the block
+#   end
+#   results.size == 5 # true
```
Comment (donaldong) [inline lib/redis_memo/batch.rb:22]:
```suggestion
#   results # [1,2,3,4,5] (results from the memoized_calculation calls)
```

File: lib/redis_memo/memoize_method.rb [hunk 4]
```diff
@@ -8,6 +8,27 @@
...
+  #     memoize_method :display_title do |post|
+  #       depends_on Post.where(id: post.id)
+  #       depends_on User.where(id: post.author_id)
+  #     end
+  #   end
+  #
+  # @param method_name [String] The name of the method to memoize
+  # @param method_id [String] Optionally, a method_id that's used to tag APM traces of RedisMemo calls.
+  # @param options [Hash] Cache options to pass to RedisMemo. These values will override the global
+  #        cache options.
```
Comment (donaldong) [inline lib/redis_memo/memoize_method.rb:30]: I think we can add some more details here: What are the supported options here? (not all options from options.rb are possible).

File: lib/redis_memo/memoize_method.rb [hunk 5]
```diff
@@ -8,6 +8,27 @@
...
+  # @param depends_on [block] The method's dependency block
```
Comment (donaldong) [inline lib/redis_memo/memoize_method.rb:31]: This could also use a bit more details: How does one use the `depends_on` API? What does this method expect?

**Review by donaldong — COMMENTED** [review]
Thanks!! Some parts could use more detailed explanations. But in general this looks good to me.

File: lib/redis_memo/memoize_method.rb (hunk 4 above)
Comment (katyho) [reply lib/redis_memo/memoize_method.rb:30]: I added some details, but options are also passed along to the Rails cache, so I didn't explicitly list the possible values there.

cc @Dingying0410  You'll likely want to add the cache validation sample rate when you merge #78

**Review by donaldong — APPROVED** [review]
Nice!

**Pull request merged**